tests/test_base.py

```diff
@@ -583,14 +578,9 @@ async def coro():
         self.assertIsNone(self.loop.get_task_factory())
 
     def _compile_agen(self, src):
-        try:
-            g = {}
-            exec(src, globals(), g)
-        except SyntaxError:
-            # Python < 3.6
-            raise unittest.SkipTest()
-        else:
-            return g['waiter']
+        g = {}
```

Comment:
I don't think this function is needed now; just use async gens syntax directly in tests.